### PR TITLE
ddb: implement transactions;

### PR DESCRIPTION
**Pull request by selm0**
AWS DynamoDB supports transactions since early 2020.
This PR adds basic support for the different methods.

### CODE REVIEW

File: pkg/ddb/builder.go [hunk 1]
```diff
@@ -32,7 +32,7 @@ func (b *keyBuilder) buildKey(item interface{}) (KeyValues, error) {
 
 func (b *keyBuilder) fromItem(item interface{}) (KeyValues, error) {
 	if item == nil {
-		return nil, fmt.Errorf("can not build key attributes from nil item")
+		return nil, fmt.Errorf("can not build key attributes from nil Item")
```
Comment (j4k4) [inline pkg/ddb/builder.go:35]: is there a reason why "Item" is now ucfirst everywhere? :thinking:

File: pkg/ddb/builder_conditioncheck.go [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+
+type conditionCheckBuilder struct {
+	condition  *expression.ConditionBuilder
+	err        error
+	metadata   *Metadata
+	keyBuilder keyBuilder
+	returnType *string
+}
+
+func NewconditionCheckBuilder(metadata *Metadata) ConditionCheckBuilder {
```
Comment (j4k4) [inline pkg/ddb/builder_conditioncheck.go:30]: should be `NewConditionCheckBuilder`?

File: pkg/ddb/builder_putitem.go [hunk 3]
```diff
@@ -48,7 +50,7 @@ func (b *putItemBuilder) ReturnAllOld() PutItemBuilder {
 
 func (b *putItemBuilder) Build(item interface{}) (*dynamodb.PutItemInput, error) {
 	if b.returnType != nil && *b.returnType != dynamodb.ReturnValueNone && !isPointer(item) {
-		return nil, fmt.Errorf("the provided old value has to be a pointer")
+		return nil, errors.New("the provided old value has to be a pointer")
```
Comment (j4k4) [inline pkg/ddb/builder_putitem.go:53]: why changing this from `fmt.Errorf` to `errors.New`?  
i like having a consistent `fmt` usage. Is there a benefit despite a little performance advantage, which should be neglectable as the error should not happen in the first place? :D

File: pkg/ddb/builder.go (hunk 1 above)
Comment (selm0) [reply pkg/ddb/builder.go:35]: overeager refactoring of my IDE. Should be reverted

File: pkg/ddb/builder_putitem.go (hunk 3 above)
Comment (selm0) [reply pkg/ddb/builder_putitem.go:53]: No, it's only about using the right tool for the right purpose (and it's a little bit faster)

Comment (j4k4) [reply pkg/ddb/builder_putitem.go]: for me, consistency would be the winner here.. also one less import..  
but also no hard feelings about it, if you want to keep it like this..

**Pull request merged**